src/type.zig

```diff
@@ -261,10 +261,54 @@ pub const Type = extern union {
                 var buf_b: Payload.ElemType = undefined;
                 return a.optionalChild(&buf_a).eql(b.optionalChild(&buf_b));
             },
+            .ErrorSet => {
+                if (a.tag() == .anyerror) {
+                    if (b.tag() == .anyerror)
+                        return true
+                    else
+                        return false;
```

Comment:
This logic is not correct, because `(error{Foo} || anyerror) == anyerror` is `true`. An inferred error set could be the LHS too. And again the concept of "resolving" an error set applies here. I think error sets are one of the trickier things to get right here and I don't want to go down a path that leads us towards bugs. Probably even the first implementation of error sets should include inferred ones.